core/src/main/scala/org/apache/spark/sql/delta/actions/DeletionVectorDescriptor.scala

```diff
@@ -0,0 +1,254 @@
+     */
+    @JsonDeserialize(contentAs = classOf[java.lang.Integer])
+    offset: Option[Int] = None,
+    /** Size of the serialized DV in bytes (raw data size, i.e. before base85 encoding). */
+    sizeInBytes: Int,
+    /** Number of rows the DV logically removes from the file. */
+    cardinality: Long) {
+
+  import DeletionVectorDescriptor._
+
+  @JsonIgnore
+  @transient
+  lazy val uniqueId: String = {
+    offset match {
+      case Some(offset) => s"$uniqueFileId@$offset"
+      case None => uniqueFileId
+    }
+  }
+
+  @JsonIgnore
+  @transient
+  lazy val uniqueFileId: String = s"$storageType$pathOrInlineDv"
+
+  @JsonIgnore
+  protected[delta] def isOnDisk: Boolean = !isInline
+
+  @JsonIgnore
+  protected[delta] def isInline: Boolean = storageType == INLINE_DV_MARKER
+
+  @JsonIgnore
+  protected[delta] def isRelative: Boolean = storageType == UUID_DV_MARKER
+
+  @JsonIgnore
+  protected[delta] def isAbsolute: Boolean = storageType == PATH_DV_MARKER
+
+  @JsonIgnore
+  protected[delta] def isEmpty: Boolean = cardinality == 0
+
+  def absolutePath(tableLocation: Path): Path = {
+    require(isOnDisk, "Can't get a path for an inline deletion vector")
+    storageType match {
+      case UUID_DV_MARKER =>
+        // If the file was written with a random prefix, we have to extract that,
+        // before decoding the UUID.
+        val randomPrefixLength = pathOrInlineDv.length - Codec.Base85Codec.ENCODED_UUID_LENGTH
+        val (randomPrefix, encodedUuid) = pathOrInlineDv.splitAt(randomPrefixLength)
+        val uuid = Codec.Base85Codec.decodeUUID(encodedUuid)
+        assembleDeletionVectorPath(tableLocation, uuid, randomPrefix)
+      case PATH_DV_MARKER =>
+        // Since there is no need for legacy support for relative paths for DVs,
+        // relative DVs should *always* use the UUID variant.
+        val parsedUri = new URI(pathOrInlineDv)
+        assert(parsedUri.isAbsolute, "Relative URIs are not supported for DVs")
+        new Path(parsedUri)
+      case _ => throw DeltaErrors.cannotReconstructPathFromURI(pathOrInlineDv)
+    }
+  }
+
+  /**
+   * Produce a copy of this DV, but using an absolute path.
+   *
+   * If the DV already has an absolute path or is inline, then this is just a normal copy.
+   */
+  def copyWithAbsolutePath(tableLocation: Path): DeletionVectorDescriptor = {
+    storageType match {
+      case UUID_DV_MARKER =>
+        val absolutePath = this.absolutePath(tableLocation)
+        this.copy(storageType = PATH_DV_MARKER, pathOrInlineDv = absolutePath.toString)
+      case PATH_DV_MARKER | INLINE_DV_MARKER => this.copy()
+    }
+  }
+
+  /**
+   * Produce a copy of this DV, with `pathOrInlineDv` replaced by a relative path based on `id`
+   * and `randomPrefix`.
+   */
+  def copyWithNewRelativePath(id: UUID, randomPrefix: String): DeletionVectorDescriptor = {
+    this.copy(storageType = UUID_DV_MARKER, pathOrInlineDv = encodeUUID(id, randomPrefix))
+  }
```

Comment:
This should be a noop copy for inline DVs, since we shouldn't write them out to disk ever anyway.

Reply:
makes sense. Will update it.